src/TestFramework/Factory.php

```diff
@@ -105,19 +106,33 @@ public function create(string $adapterName, bool $skipCoverage): TestFrameworkAd
             );
         }
 
-        if ($adapterName === TestFrameworkTypes::CODECEPTION) {
-            $codeceptionConfigPath = $this->configLocator->locate(TestFrameworkTypes::CODECEPTION);
+        $extensionsClassExists = class_exists('\Infection\ExtensionInstaller\GeneratedExtensionsConfig');
 
-            return CodeceptionAdapterFactory::create(
-                (new TestFrameworkFinder('codecept'))->find(),
-                $this->tmpDir,
-                $codeceptionConfigPath,
-                null,
-                $this->jUnitFilePath,
-                $this->projectDir,
-                $this->infectionConfig->getSourceDirectories(),
-                $skipCoverage
-            );
+        if ($extensionsClassExists) {
+            $installedExtensions = \Infection\ExtensionInstaller\GeneratedExtensionsConfig::EXTENSIONS;
+
+            foreach ($installedExtensions as $packageName => $installedExtension) {
+                $factory = $installedExtension['extra']['class'];
+
+                Assert::classExists($factory);
+
+                if (!is_a($factory, TestFrameworkAdapterFactory::class, true)) {
```

Comment:
shouldn't that throw an error instead?

Reply:
No, because `extension-installer` is not only about about Test framework Adapters. It can autodiscover any plugin with type `infection-extension` (in the future it will be any extension point).

That's why here we have to filter out not needed classes.

We can't do filtering in the `extension-installer` plugin itself because in this case plugin will depend on Test Framework Adapter, Mutator, etc. which is not what is created for.